### PR TITLE
added online pagination - infrastructure & logic & example

### PR DESCRIPTION
**Pull request by itaykl**

### CODE REVIEW

File: packages/polaris-test/server/schema/resolvers.ts [hunk 1]
```diff
@@ -118,6 +120,44 @@ export const resolvers = {
         customContextInstanceMethod: (parent: any, args: any, context: TestContext): string =>
             context.instanceInContext.doSomething(),
         permissionsField: () => 'foo bar baz',
+        onlinePaginatedBooks: async (
+            parent: any,
+            args: any,
+            context: TestContext,
+        ): Promise<PageConnection<Book>> => {
+            const connection = getPolarisConnectionManager().get(process.env.SCHEMA_NAME);
+            let books = await connection.getRepository(Book).find(context);
```
Comment (chenshoo) [inline packages/polaris-test/server/schema/resolvers.ts:130]: sort with typeorm

File: packages/polaris-test/server/schema/resolvers.ts [hunk 2]
```diff
@@ -118,6 +120,44 @@ export const resolvers = {
...
+            books.sort((book1, book2) => (book1.getId() > book2.getId() ? 1 : -1));
+            const copyOfBooks = books;
```
Comment (chenshoo) [inline packages/polaris-test/server/schema/resolvers.ts]: check if not a reference, add tests, add read me, add types to nest

**Review by osher-sade — APPROVED** [review]

File: packages/polaris-schema/README.md [hunk 3]
```diff
@@ -29,6 +29,24 @@ It consists the following properties:
...
+#### PageConnection<ENTITY>
+
+This interface represents generic connection type(of type <ENTITY>) to use when implementing an online pagination - it
+contains `pageInfo` and `edges` fields which determines our page structure and data it will contain.
+Below will be more explanation about these fields and usage.
+
+#### PageInfo
+
+This interface represents a metadata-like information about our current page when executing an online pagination.
+It may contains information like `startCursor` and `endCursor` of our page and wether it `hasNextPage` and `hasPreviousPage`.
```
Comment (chenshoo) [inline packages/polaris-schema/README.md]: whether

File: packages/polaris-schema/README.md [hunk 4]
```diff
@@ -38,7 +56,89 @@ explained above.
 
 All of the scalars supported by polaris-schema.
 
+#### PageInfoTypeDefs
+
+The `PageInfo` typedefs provides a metadata information when executing an online pagination.
+It contains `startCursor` and `endCursor` which indicates the current page limits.
+In general the cursor's job is to determine by which field our paging ........
```
Comment (chenshoo) [inline packages/polaris-schema/README.md]: ....?

File: packages/polaris-schema/README.md [hunk 5]
```diff
@@ -38,7 +56,89 @@ explained above.
...
+It also contains `hasNextPage` and `hasPreviousPage` indicators relative to te current page.
```
Comment (chenshoo) [inline packages/polaris-schema/README.md]: the

File: packages/polaris-test/tests/online-pagination.test.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import * as createBook from './jsonRequestsAndHeaders/createBook.json';
+import * as onlineBooksPagination from './jsonRequestsAndHeaders/onlineBooksPagination.json';
+
+const setUp = async (): Promise<any[]> => {
+    const bookIds: any[] = [];
+    for (let i = 1; i <= 10; i++) {
+        const res = await graphQLRequest(createBook.request, {}, { title: `book${i}` });
+        bookIds.push(res.createBook.id);
+    }
+    return bookIds;
```
Comment (chenshoo) [inline packages/polaris-test/tests/online-pagination.test.ts]: you don't really use the bookIds, why return them and why do tear down? every test you have a new server up

Comment (itaykl) [reply packages/polaris-test/tests/online-pagination.test.ts]: you are correct, i forgot it there.. good comment

**Pull request merged**